### ISSUE TITLE
Build-enabled pipelines do not work in hub

### ISSUE THREAD

**Issue by ysimonson** [pachyderm/pachyderm#5141]
(I'm not sure if this is a core issue or hub issue.)

**What happened?**: Build-step enabled pipelines do not work in hub. This is because they first create a repo, then create a pipeline on that existing repo, and this triggers a permissions error - even though this works fine on clusters deployed elsewhere.

**How to reproduce it (as minimally and precisely as possible)?**: Run `examples/word_count` on hub. It will trigger this error when trying to create the map pipeline:

```
failed to create build pipeline for build step-enabled pipeline: cannot overwrite repo "map_build" with new output repo
```

**Comment by msteffen** [comment]
Sorry, this issue was robo-closed by GitHub, but there's another change we need to make in CreatePipeline before this is fixed

**Comment by ysimonson** [comment]
That's my bad! I attached the issue to the PR thinking it was definitely going to fix it.

**Comment by PFedak** [comment]
(Should have been closed by #5594, whoops)
